devices/HumanStateProvider/InverseVelocityKinematics/InverseVelocityKinematics.cpp

```diff
@@ -8,22 +8,55 @@
 
 #include "InverseVelocityKinematics.hpp"
 
-#include <Eigen/QR>
-#include <Eigen/SparseCholesky>
-#include <iDynTree/Core/EigenHelpers.h>
-#include <iDynTree/Core/VectorDynSize.h>
-#include <iDynTree/KinDynComputations.h>
+// osqp-eigen
+#include "OsqpEigen/OsqpEigen.h"
 
 // ====
 // IMPL
 // ====
 
 class InverseVelocityKinematics::impl
 {
+    // qp variables
+    iDynTree::VectorDynSize m_u_prime, m_l_prime, m_q_prime;
+    iDynTree::MatrixDynSize m_A_prime, m_P_prime;
+
+    iDynTree::VectorDynSize m_u, m_l, m_q;
+    iDynTree::MatrixDynSize m_A, m_P;
+
+    std::unique_ptr<OsqpEigen::Solver> m_optimizerSolver; /**< Optimization solver. */
+
 public:
     iDynTree::Model model;
     iDynTree::KinDynComputations dynamics;
-    size_t dofs;
+    size_t dofs; // joint configuration size
+    size_t configSpaceSize; // configuration space DoFs (dofs+ baseDofs)
+    size_t baseDofs; // base DoFs
+
+    //*********** custom cosntraints *************//
```

Comment:
duplicated title, to be removed from here

Reply:
I don't get, what  is duplicated?

Reply:
this title is actually placed also below (https://github.com/robotology/human-dynamics-estimation/pull/137/files/e21b51e4d984cad817c33849305240f72dadf57e#diff-f8d68d61a7d1cf3627cbd20c215bb5a8R41), while here are the variables related to `general constraints` as the title below indicates (https://github.com/robotology/human-dynamics-estimation/pull/137/files/e21b51e4d984cad817c33849305240f72dadf57e#diff-f8d68d61a7d1cf3627cbd20c215bb5a8R37)

Reply:
This line still needs to be removed